src/compiler/scala/tools/nsc/Driver.scala

```diff
@@ -18,7 +18,8 @@ abstract class Driver {
   }
 
   protected def processSettingsHook(): Boolean = {
-    if (settings.version) { reporter echo versionMsg ; false } else true
+    if (settings.version) { reporter echo versionMsg ; false }
+    else !reporter.hasErrors
```

Comment:
I'll have to view full file to answer that -- is it the comment on line 42, there is a reporter for early errors before we lock and load the compiler's reporter on line 48.